wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/LogFilesProvider.swift

```diff
@@ -0,0 +1,174 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import UIKit
+import WireCommonComponents
+import WireSystem
+import ZipArchive
+
+// sourcery: AutoMockable
+protocol LogFilesProviding {
+
+    /// Generates a zip file containing all log files and returns its data before removing the files
+    ///
+    /// - Returns: the log files archive data
+
+    func generateLogFilesData() throws -> Data
+
+    /// Generates a zip file containing all log files
+    /// 
+    /// - Returns: the log files archive URL
+
+    func generateLogFilesZip() throws -> URL
+
+    /// Clears the logs directory.
+    /// Call once you are done using the URL returned by `generateLogFilesZip` to clean up.
+
+    func clearLogsDirectory() throws
+
+}
+
+struct LogFilesProvider: LogFilesProviding {
```

Comment:
So to recap:
- all logs are now stored in the tmp folder of the user in /randomUUID/logs
- when sending logs, we create a unique folder for the zip then delete in /randomUUID/logs/UUID/logs.zip
- we add info.txt to contain general info inside the logs.

Can we sum up this info inside docC to wrap this all up?

Reply:
https://github.com/wireapp/wire-ios/pull/1658/commits/c81926da4e3baa2bf5253ca1bc051109d41a4c71